consensus/src/lib.rs

```diff
@@ -100,6 +100,24 @@ cfg_if::cfg_if! {
                 context: Self::Context,
                 payload: Self::Digest,
             ) -> impl Future<Output = oneshot::Receiver<bool>> + Send;
+
+            /// Determine whether a notarized payload is safe to finalize.
+            ///
+            /// Block-coding applications can override this to delay or prevent
+            /// finalization until they have reconstructed and validated the full
+            /// block (e.g., after receiving enough shards).
```

Comment:
nit: Elaborate that this provides a way to exclude a notarized block from the canonical chain.